### PR TITLE
Feature login

### PR DESCRIPTION
**Pull request by SabinePfanner**
Enable login functionality via GitHub and block several features for not logged in users

US: https://github.com/users/SabinePfanner/projects/2/views/1?pane=issue&itemId=65503971

### CODE REVIEW

File: components/Login.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      </>
+    );
+  }
+  return (
+    <>
+      <StyledLoginButton onClick={() => signIn()}>
+        <SvgIcon variant="login" size="30" color="#1D0B07" />
+      </StyledLoginButton>
+    </>
+  );
```
Comment (nichoop) [inline components/Login.js:28]: The fragment can be omitted as the StyledLoginButton is the only element being returned in both cases.

File: pages/api/auth/[...nextauth].js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          },
+        })
+      : GithubProvider({
+          clientId: process.env.GITHUB_ID,
+          clientSecret: process.env.GITHUB_SECRET,
+        }),
+    // ...add more providers here
+  ],
+};
+
```
Comment (nichoop) [inline pages/api/auth/[...nextauth].js:40]: Just a curiosity: is it possible here to add an error message when the authentication does not work?

**Review by nichoop — APPROVED** [review]
Looks great!
I've just added 2 comments :)

File: pages/api/auth/[...nextauth].js (hunk 2 above)
Comment (SabinePfanner) [reply pages/api/auth/[...nextauth].js:40]: Good question - I guess this is handled automatically be the authentication provider. But will keep that in mind for the future, maybe we implement something like "Login not possible"

**Pull request merged**